stix2/datastore/filesystem.py

```diff
@@ -77,15 +426,23 @@ def stix_dir(self):
     def _check_path_and_write(self, stix_obj):
         """Write the given STIX object to a file in the STIX file directory.
         """
-        path = os.path.join(self._stix_dir, stix_obj["type"], stix_obj["id"] + ".json")
+        type_dir = os.path.join(self._stix_dir, stix_obj["type"])
+        if is_marking(stix_obj):
+            filename = stix_obj["id"]
+            obj_dir = type_dir
+        else:
+            filename = _timestamp2filename(stix_obj["modified"])
+            obj_dir = os.path.join(type_dir, stix_obj["id"])
+
+        file_path = os.path.join(obj_dir, filename + ".json")
 
-        if not os.path.exists(os.path.dirname(path)):
-            os.makedirs(os.path.dirname(path))
+        if not os.path.exists(obj_dir):
+            os.makedirs(obj_dir)
 
         if self.bundlify:
             stix_obj = Bundle(stix_obj, allow_custom=self.allow_custom)
 
-        with open(path, "w") as f:
+        with open(file_path, "w") as f:
             f.write(str(stix_obj))
```

Comment:
I know this was unchanged, but str(stix_obj) is not the right way to serialize objects. You could do something like [this](https://github.com/oasis-open/cti-python-stix2/blob/stix2.1/stix2/datastore/filesystem.py#L93-L95)

Reply:
Responding to you questions. The tool when you run it locally will tell you which files failed the test and fix them for you. So, the files you see `"Fixing stix2/test/..."` are the problematic ones. The objective of the hook is to prevent extra/unnecessary white-space on the code, but since we pass a "--all-files" flag I think it also consider non-code files like .json.

Reply:
We don't have to fix the str(stix_obj) thing here since I already have fixes on my other PR branch. It can be taken care on the upcoming merges.